Public/Src/Engine/Dll/EngineSchedule.cs

```diff
@@ -771,21 +771,46 @@ private bool ShouldSerializeOptimizationDataStructurePostExecution()
                 outputDirectories = scheduler.PipGraph.AllDirectoriesContainingOutputs().Select(d => d.ToString(scheduler.Context.PathTable)).ToList();
             }
 
+            var scrubber = new DirectoryScrubber(
+                cancellationToken: scheduler.Context.CancellationToken,
+                loggingContext: loggingContext,
+                loggingConfiguration: configuration.Logging,
+                maxDegreeParallelism: Environment.ProcessorCount,
+                tempDirectoryCleaner: tempCleaner);
+
+            var journalFiles = SharedOpaqueJournal
+                .FindAllJournalFiles(configuration.Layout.SharedOpaqueJournalDirectory.ToString(scheduler.Context.PathTable))
+                .ToArray();
+            var distinctRecordedWrites = journalFiles
+                .AsParallel()
+                .WithDegreeOfParallelism(Environment.ProcessorCount)
+                .WithCancellation(scheduler.Context.CancellationToken)
+                .SelectMany(SharedOpaqueJournal.ReadRecordedWritesFromJournalWrapExceptions)
+                .Distinct(StringComparer.OrdinalIgnoreCase)
+                .Where(FileUtilities.FileExistsNoFollow)

```

Comment:
>.Where(FileUtilities.FileExistsNoFollow) [](start = 16, length = 40)

And a similar comment here. Do you actually need to probe every single file? You could try to unconditionally delete it, and if the file is not there, we are good as well

Reply:
will try

Reply:
removed probing from here